core/consensus/safety/src/lib.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Common utilities for accountable safety in Substrate.
+
+#![forbid(missing_docs, unsafe_code)]
+
+use client;
+use transaction_pool::txpool::{self, PoolApi};
+use node_runtime::{UncheckedExtrinsic, Call};
+use parity_codec::{Encode, Decode};
+use std::sync::Arc;
+use runtime_primitives::traits::{Block as BlockT};
+use runtime_primitives::generic::BlockId;
+use log::{error, warn, debug, info, trace};
+use client::blockchain::HeaderBackend;
+
+/// Submit report call to the transaction pool.
+pub fn submit_report_call<C, T, Block>(
+	client: &Arc<C>,
+	transaction_pool: &Arc<T>,
```

Comment:
These seem like they could both be `&X` instead of `&Arc<X>`. Why leak `Arc` into it?